packages/block-editor/src/components/editor-styles/index.js

```diff
@@ -18,53 +18,49 @@ import { useSelect } from '@wordpress/data';
 import transformStyles from '../../utils/transform-styles';
 import { store as blockEditorStore } from '../../store';
 import { unlock } from '../../lock-unlock';
+import useEditorFontsResolver from '../use-editor-fonts-resolver';
 
 extend( [ namesPlugin, a11yPlugin ] );
 
-function useDarkThemeBodyClassName( styles, scope ) {
-	return useCallback(
-		( node ) => {
-			if ( ! node ) {
-				return;
-			}
+function useDarkThemeBodyClassName( scope, ref ) {
```

Comment:
Why did you refactor this from a hook that returns a ref to a hook that receives a ref as an argument?

Returning a ref allow supporting conditional rendering, you can adapt the hook when the "node" changes, it's not something you can do in hooks that receive refs. 

Receiving refs in hooks is a pattern we should avoid (See recent discussion here https://github.com/WordPress/gutenberg/pull/64943#discussion_r1741052415 )

Reply:
> Why did you refactor this from a hook that returns a ref to a hook that receives a ref as an argument?

Because it's hard to understand the pattern (at least for me :laughing: ) and what it does, so I thought it wasn't something 100% intentional. Thanks for linking providing context. I see the reasons why it is like that now. I reverted the changes to that hook and refactored the new useEditorFontsResolver to follow that pattern.

https://github.com/WordPress/gutenberg/pull/65019/commits/463fa952b9cb2170e89398a36625f4195faca6eb